opentracing/scope_managers/asyncio.py

```diff
@@ -20,32 +20,29 @@
 
 from __future__ import absolute_import
 
-import asyncio
+from contextvars import ContextVar
 
 from opentracing import Scope
 from opentracing.scope_managers import ThreadLocalScopeManager
-from .constants import ACTIVE_ATTR
+
+
+_SCOPE = ContextVar('scope')
 
 
 class AsyncioScopeManager(ThreadLocalScopeManager):
     """
     :class:`~opentracing.ScopeManager` implementation for **asyncio**
-    that stores the :class:`~opentracing.Scope` in the current
-    :class:`Task` (:meth:`Task.current_task()`), falling back to
-    thread-local storage if none was being executed.
+    that stores the :class:`~opentracing.Scope` using ContextVar.
 
-    Automatic :class:`~opentracing.Span` propagation from
-    parent coroutines to their children is not provided, which needs to be
-    done manually:
+    The scope manager provides automatic :class:`~opentracing.Span` propagation
```

Comment:
This is one reason that makes me feel we should keep this new implementation under `ContextVarScopeManager` or similar - otherwise, we would be breaking the assumption that manual propagation is required. 

At the same time, does `aiocontextvars` work for Python 3.4/3.5? If not, we cannot use them with `asyncio` (I bring this up as I've talked to people and the need to still keep Python 3.4 backwards compatibility).

Reply:
Hello @carlosalberto !

> ...otherwise, we would be breaking the assumption that manual propagation is required (we could at the same time deprecate AsyncioScopeManager in favor of this new one, of course).

That's what I meant about backward compatibility of new scope manager :) No need to activate parent span, but you can do it and it doesn't break behavior of existing user code. There are tests that prove it.

> I bring this up as I've talked to people and the need to still keep Python 3.4 backwards compatibility

Wow! But this is really strong reason to separate new scope manager, because both libraries (`contextvars` and `aiocontextvars`) doesn't support python 3.4. So this brave new `ContextVarScopeManager` we have to use only with `python >= 3.5`

> we could at the same time deprecate AsyncioScopeManager in favor of this new one, of course

Maybe we can release new scope manager in new major version of the library (3.0.0) with renamed existing `AsyncioScopeManager` -> `OldAsyncioScopeManager`? Yes, it will break backward compatibility for python 3.4 users, but it's ok in terms of semver.